Content.Server/GameObjects/Components/Atmos/GasTankComponent.cs

```diff
@@ -1,10 +1,110 @@
-﻿using Robust.Shared.GameObjects;
+﻿using System;
+using Content.Server.Atmos;
+using Content.Server.GameObjects.Components.Items.Storage;
+using Content.Server.Interfaces;
+using Content.Shared.GameObjects;
+using Content.Shared.GameObjects.EntitySystems;
+using Content.Shared.GameObjects.Verbs;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Content.Shared.Utility;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+using Robust.Shared.ViewVariables;
 
 namespace Content.Server.GameObjects.Components.Atmos
 {
     [RegisterComponent]
-    public class GasTankComponent : Component
+    public class GasTankComponent : ItemComponent, IExamine, IGasMixtureHolder
     {
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+
+        private const float MinimumThreshold = 16f;
+        private const float DefaultOutputRatio = 0.01f;
         public override string Name => "GasTank";
+        public override uint? NetID => ContentNetIDs.GAS_TANK;
+
+        [ViewVariables] public GasMixture Air { get; set; }
+        [ViewVariables] public bool IsOpen { get; set; }
+        [ViewVariables] public float OutputRatio { get; set; }
```

Comment:
@Zumorica Do you mean we need to have maximum Output kPA instead of ratio? and try to equalize the tank contents with the outside environment during deltaTime?